Sources/ValidEmail.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  ValidEmail.swift
+//  SwiftPlusPlus
+//
+//  Created by Eric Dockery on 4/7/17.
+//  Copyright © 2017 Drewag. All rights reserved.
+//
+
+import Foundation
+
+struct ValidEmail {
+    public init(string: String) throws {
+        guard string.isValidEmail else {
+            throw LocalUserReportableError(
```

Comment:
1. Let's call this `EmailAddress` instead of ValidEmail
2. This type needs to be public
3. You need to store the string and make it available as a property. It think `raw` is a good property name. (also make sure the property is public).
4. It should implement the CustomStringConvertible